flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/optimize/OptimizerBase.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.plan.optimize
+
+import org.apache.flink.table.plan.schema.IntermediateRelTable
+
+import org.apache.calcite.rel.core.TableScan
+import org.apache.calcite.rel.{RelNode, RelShuttleImpl}
+
+import java.util
+
+import scala.collection.JavaConversions._
+
+/**
+  * Base class for [[Optimizer]].
+  */
+abstract class OptimizerBase extends Optimizer {
```

Comment:
I'm not sure this is some kind of base optimizer for all the optimizers. It provides some specific functionality with `IntermediateRelNodeTable`, and not every optimizer should have this concept. Maybe you can change the name to a more dedicated form.

Reply:
yes, i think `CommonSubGraphBasedOptimizer` is better.